louisewoodhams/LOUISE_WOODHAMS.md

```diff
@@ -0,0 +1,78 @@
+
+With API testing you'll probably be writing a bunch of JSON. If you haven't had much exposure to coding at this point you'll probably have the problem I had which was you'll mess up your braces {} or miss off a comma or a bracket [] for an array as you just aren't use to using them or you copy pasted from Notepad++ and missed a bit off. This is one of the places Swagger falls down as well. Within the "Try It Out" boxes with the example code you can remove things by accident when adding in your test responses and it will just give you a 400: Bad Request with little to no reason as to why. It is really hard to work out where you've gone wrong and sometimes it is because you've 
+copy and pasted in from a PBI/Story/item and it has added a bunch of random spaces which you can't see. I have a tip for you here. Copy paste it into Postman or VS code. Both will tell you where you've gone wrong and highlight where you need to fix things. If you still get a 400 after this then what you are sending as data might not be right.
+
+
+## Validation And Required Fields 
+
+Validation and Mandatory fields / boxes will be a thing you will have probably come across during your testing journey or just in you day to day with the apps you use or rely on. In my experience however they do tend to get forgotten about by developers when they knock up an API. Typically not supplying things for mandatory / required fields should give you a ticket to 400: Bad Request Ville but this is usually a great opportunity to cause some explosions and get some 500s where validation has been missed or you're sending something unexpected.
+
+## HTTP Codes And Checking For Teapots 
+
+I'm a major geek about HTTP Codes because of the API testing I've done over the years. The full list of there is here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status and I'll explain a couple that 
+have brought me joy and sadness. Most of the time you will want to be getting things in the 200 range depending if data is coming back or not and depending on the kind of data that was supplied in the first place.
+I haven't encountered many 300s in my API testing time but I haven't tested a massive range of APIs. The 300 range deals with redirection. The 400 range deals with errors. This can either be caused by a lack 
+of permission your side in relation to what you are sending or a lot of the time your authentication has timed out but sometimes the admin you are authenticated as doesn't have permission to do that (a DELETE for example). Other
+times it is due to the Request type being wrong, you could be sending your request as a GET and it is supposed to be a POST 405: Method Not Allowed. My favourite of the 400s is 418: I'm A Teapot and 
+I recommend if you are involved in building an internal REST API you encourage your development team to build in a legit 418, bonus points for custom error messages and Postman tests!
+
+ ## The Joys of 500s And How You Should Never See Them 
```

Comment:
Not coming through as a heading in my viewer. Indentation issue maybe?

Reply:
Seems to be fine locally and in the github view, do you have a screenshot?

Reply:
Hmm maybe its just when I view the file 
![image](https://user-images.githubusercontent.com/16778356/111366074-f80ad380-868a-11eb-9eeb-ccbf50aadaa1.png)